salt/utils/napalm.py

```diff
@@ -281,7 +283,24 @@ def get_device(opts, salt_obj=None):
     '''
     log.debug('Setting up NAPALM connection')
     network_device = get_device_opts(opts, salt_obj=salt_obj)
-    _driver_ = napalm_base.get_network_driver(network_device.get('DRIVER_NAME'))
+    provider_lib = napalm_base
+    if network_device.get('PROVIDER'):
+        # In case the user requires a different provider library,
+        #   other than napalm-base.
+        # For example, if napalm-base does not satisfy the requirements
+        #   and needs to be enahanced with more specific features,
+        #   we may need to define a custom library on top of napalm-base
+        #   with the constraint that it still needs to provide the
+        #   `get_network_driver` function. However, even this can be
+        #   extended later, if really needed.
+        # Configuration example:
+        #   provider: napalm_base_example
+        try:
+            provider_lib = importlib.import_module(network_device.get('PROVIDER'))
+        except ImportError as ierr:
+            log.error('Unable to import {0}'.format(network_device.get('PROVIDER')), exc_info=True)
+            log.error('Falling back to napalm-base')
+    _driver_ = provider_lib.get_network_driver(network_device.get('DRIVER_NAME'))
```

Comment:
Should we check first to see if `get_network_driver()` exists in the module, or just let it raise AttributeError?